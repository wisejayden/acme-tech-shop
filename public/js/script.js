(function() {

		var container = document.getElementById( 'container' ),
			trigger = container.querySelector( 'button.catalog' );


		function toggleContent() {
			if( classie.has( container, 'container--open' ) ) {
				classie.remove( container, 'container--open' );
				classie.remove( trigger, 'trigger--active' );
				window.addEventListener( 'scroll', noscroll );
			}
			else {
				classie.add( container, 'container--open' );
				classie.add( trigger, 'trigger--active' );
				window.removeEventListener( 'scroll', noscroll );
			}
		}

		function noscroll() {
			window.scrollTo( 0, 0 );
		}

		// reset scrolling position
		document.body.scrollTop = document.documentElement.scrollTop = 0;

		// disable scrolling
		window.addEventListener( 'scroll', noscroll );

		trigger.addEventListener( 'click', toggleContent {
			console.log("HELLo")
		} );

	})();
