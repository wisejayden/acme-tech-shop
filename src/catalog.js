import React from 'react';
import axios from './axios';
import {Link} from 'react-router-dom';



export class Catalog extends React.Component {
    constructor(props) {
        super(props);
        this.state= {
            monoqiAPI: 'http://challenge.monoqi.net/catalog',
            mockAPI: '/catalog/api'
        };
    }
    componentDidMount() {
        //On mount, retrieve entire catalog from server and then dynamically add to the DOM
        axios.get(this.state.mockAPI)
            .then((res) => {
                var articles = res.data.articles;
                console.log("articles", articles);
                console.log("resdata", res.data);
                let allArticles = [];
                for (var i = 0; i < articles.length; i++) {
                    var sku = articles[i].sku;
                    allArticles.push(
                            <div key={i} className="catalog-item-container">
                                <Link to={"/item/" + articles[i].sku}><img name={articles[i].sku} className= "catalog-item-picture" src={articles[i].image} />
                                <div className="catalog-middle">
                                    <div className="catalog-text">Show More</div>
                                </div></Link>
                                <Link to={"/item/" + articles[i].sku}><h2>{articles[i].name}</h2></Link>
                                <p>{articles[i].price.amount} {articles[i].price.currency}</p>
                            </div>
                    )
                }
                this.setState({
                    allArticles
                })
            })
            .catch(err => {
                console.log("err", err);
            })
    }

    render() {
        return(
            <div className="catalog">
                {this.state.allArticles}
            </div>
        )
    }
}
