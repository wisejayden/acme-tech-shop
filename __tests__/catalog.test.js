import React from 'react';
import { shallow, mount, render } from 'enzyme';
import {Catalog} from '../src/catalog.js';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';


test('Catalog component should render as expected', () => {
    const component = shallow(<Catalog />);
    const tree = toJson(component);
    expect(tree).toMatchSnapshot();
})
