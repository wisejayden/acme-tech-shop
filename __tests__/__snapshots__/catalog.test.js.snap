// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Catalog component should render as expected 1`] = `
<div
  className="catalog"
/>
`;
